test/fake_http_server.h

```diff
@@ -31,15 +32,45 @@ class FakeServer {
   std::string GetUrl();
   void SetResponse(const std::string& path, const std::string& response);
 
+  // Indicates that for the given path, the server should stream
+  // responses over a hanging GET.
+  void AllowStream(const std::string& path);
+
+  // Blocks until at least one client has connected to the given path.
+  // Returns false if no client has connected after 3 seconds.
+  bool WaitForOneStreamWatcher(const std::string& path);
+
+  // Sends a streaming response to all watchers fo the given path.
+  void SendStreamResponse(const std::string& path, const std::string& response);
+
+  // Closes all open streams on the server.
+  void TerminateAllStreams();
+
  private:
   struct Handler;
   typedef boost::network::http::server<Handler> Server;
 
   // Handler that maps paths to response strings.
   struct Handler {
+    // Stream holds the state for an endpoint that streams data over a
+    // hanging GET.
+    //
+    // There is a queue for each client, and the mutex guards access
+    // to queues vector and any queue element.
+    //
+    // This struct only holds a pointer to each queue; it does not
+    // take ownership.
+    struct Stream {
+      std::mutex mutex;
+      std::condition_variable cv;
+      std::vector<std::queue<std::string>*> queues;
```

Comment:
I would add a comment here that says "Not owned". Those raw pointers confused me a bit.

Reply:
Done.